src/Apps/Order/__tests__/OrderApp.test.tsx

```diff
@@ -1,16 +1,145 @@
+    const { redirect } = await render("/order2/1234/shipping", {
+      Order: () => ({
+        id: 1234,
+        state: "PENDING",
+        requestedFulfillment: {
+          __typename: "Pickup",
+        },
+      }),
+    })
+    expect(redirect).toBe(undefined)
+  })
+  it("redirects to the status route if the order is not pending", async () => {
+    const { redirect } = await render("/order2/1234/shipping", {
+      Order: () => ({
+        id: 1234,
+        state: "ABANDONED",
+        requestedFulfillment: {
+          __typename: "Pickup",
+        },
+      }),
+    })
+    expect(redirect.url).toBe("/order2/1234/status")
+  })
+
+  it("stays on the shipping route if no shipping option is set", async () => {
+    const { redirect } = await render("/order2/1234/shipping", {
+      Order: () => ({
+        id: 1234,
+        state: "PENDING",
+        requestedFulfillment: null,
+      }),
+    })
+    expect(redirect).toBe(undefined)
+  })
+
+  it("redirects to the shipping route from the payment route if no shipping option was set", async () => {
+    const { redirect } = await render("/order2/1234/payment", {
+      Order: () => ({
+        id: 1234,
+        state: "PENDING",
+        requestedFulfillment: null,
+      }),
+    })
+    expect(redirect.url).toBe("/order2/1234/shipping")
+  })
+
+  it("stays on the payment route if there is shipping but no payment info", async () => {
+    const { redirect } = await render("/order2/1234/payment", {
+      Order: () => ({
+        id: 1234,
+        state: "PENDING",
+        requestedFulfillment: {
+          __typename: "Ship",
+        },
+        creditCard: null,
+      }),
+    })
+    expect(redirect).toBe(undefined)
+  })
+
+  it("redirects to the shipping route from the review route if no shipping option was set", async () => {
+    const { redirect } = await render("/order2/1234/review", {
+      Order: () => ({
+        id: 1234,
+        state: "PENDING",
+        requestedFulfillment: null,
+        creditCard: null,
+      }),
+    })
+    expect(redirect.url).toBe("/order2/1234/shipping")
+  })
+
+  it("redirects to the payment route from the review route if no credit card is set", async () => {
+    const { redirect } = await render("/order2/1234/review", {
+      Order: () => ({
+        id: 1234,
+        state: "PENDING",
+        requestedFulfillment: {
+          __typename: "Ship",
+        },
+        creditCard: null,
+      }),
+    })
+    expect(redirect.url).toBe("/order2/1234/payment")
+  })
+
+  it("stays on the review route if there are payment and shipping options set", async () => {
+    const { redirect } = await render("/order2/1234/review", {
+      Order: () => ({
+        id: 1234,
+        state: "PENDING",
+        requestedFulfillment: {
+          __typename: "Ship",
+        },
+        creditCard: {
+          id: "12345",
+        },
+      }),
+    })
+    expect(redirect).toBe(undefined)
+  })
+})
```

Comment:
Glad to see that after the `render` util all of these tests are straightforward and simple